nan.h

```diff
@@ -801,44 +801,62 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
+        v8::Isolate::GetCurrent()->GetCurrentContext()->Global() :
+            target->ToObject();
     return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+        v8::Isolate::GetCurrent(), target_, func, argc, argv));
 #else
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
+        v8::Isolate::GetCurrent()->GetCurrentContext()->Global() :
+            Nan::To<v8::Object>(target).ToLocalChecked();
     return node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv);
+        v8::Isolate::GetCurrent(), target_, func, argc, argv);
 #endif
   }
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::String> symbol
     , int argc
     , v8::Local<v8::Value>* argv) {
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
+    v8::Local<v8::Object> target_ = target->IsUndefined() || target->IsNull() ?
```

Comment:
I see it as privatish, but you are right. However, that would make the line too long so I would have to use if-else like a chump...

On October 21, 2015 7:51:59 PM GMT+03:00, Ben Noordhuis <notifications@github.com> wrote:
>>      , v8::Local<v8::String> symbol
>>      , int argc
>>      , v8::Local<v8::Value>* argv) {
>>  #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
>> +    v8::Local<v8::Object> target_ = target->IsUndefined() ||
>target->IsNull() ?
>
>target_obj?  Ending in an underscore suggests it's a member.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/499/files#r42650435